apps/volunteers/views.py

```diff
@@ -40,9 +43,17 @@ def index(request):
 )
 def profile(request, pk):
     volunteer = get_object_or_404(Volunteer, id=pk)
+    user=request.user
+    profile = Profile.objects.filter(user=user).first()
+    if profile is not None and user.desig == 'v':
+        volun = Volunteer.objects.get(profile=profile)
+    if volun is not None:
+        volun_schs = VolunteerSchedule.objects.all().filter(volun=volun).order_by('day')
+
```

Comment:
Why have you just copy pasted the code of `context_processors.py` here? Most of this code is not needed here.